src/main/java/edu/phema/elm_to_omop/vocabulary/ElmCodeResolvingValuesetService.java

```diff
@@ -0,0 +1,151 @@
+                if (PhemaVocabularyTranslator.translationSupportedForCodeSystem(code.getCodeSystem())) {
+                    translated = PhemaVocabularyTranslator.translateCode(code);
+                } else {
+                    translated = code;
+                }
+
+                if (!translatedCodes.contains(translated)) {
+                    translatedCodes.add(translated);
+                    codeOidMap.put(translated, codeDef.getName());
+                }
+            } catch (PhemaTranslationException pte) {
+                throw new ValuesetServiceException("Error translating ELM code", pte);
+            }
+        }
+
+        return translatedCodes;
+    }
+
+    private Concept getConcept(PhemaCode code) throws ValuesetServiceException {
+        List<Concept> concept = null;
+        try {
+            concept = omopService.vocabularySearch(code.getCode(), code.getCodeSystem());
+        } catch (OmopRepositoryException e) {
+            throw new ValuesetServiceException("Error searching for concept", e);
+        }
+
+        // Filter for only exact matches (this isn't possible current with the OHDSI WebAPI)
+        concept = concept.stream()
+            .filter(c -> c.conceptCode.equals(code.getCode()))
+            .collect(Collectors.toList());
+
+        if (concept.size() > 1) {
+            // The concept code is not specific enough
+            throw new ValuesetServiceException(String.format("Concept code %s does not specify a single concept in code system %s", code.getCode(), code.getCodeSystem()));
+        } else if (concept.size() == 0) {
+            // The code is missing in the OMOP instance
+            throw new ValuesetServiceException(String.format("Concept code %s does not exist in OMOP instance", code.getCode()));
+        } else {
+            return concept.get(0);
+        }
+    }
+
+    private PhemaConceptSet buildConceptSet(Concept concept, String oid) {
+        PhemaConceptSet phemaConceptSet = new PhemaConceptSet();
+
+        phemaConceptSet.setOid(oid);
+        phemaConceptSet.name = oid;
+
+        ConceptSetExpression.ConceptSetItem item = new ConceptSetExpression.ConceptSetItem();
+        item.concept = concept;
+
+        ConceptSetExpression conceptSetExpression = new ConceptSetExpression();
+        conceptSetExpression.items = new ConceptSetExpression.ConceptSetItem[]{item};
+
+        phemaConceptSet.expression = conceptSetExpression;
+        phemaConceptSet.id = conceptSetId++;
+
+        return phemaConceptSet;
+    }
+
+    @Override
+    public PhemaConceptSetList getConceptSetList() throws ValuesetServiceException {
+        PhemaConceptSetList phemaConceptSetList = new PhemaConceptSetList();
+
+        List<PhemaCode> codes = translateCodes();
+
+        Concept concept = null;
+        for (PhemaCode code : codes) {
+            concept = getConcept(code);
+            phemaConceptSetList.addConceptSet(buildConceptSet(concept, codeOidMap.get(code)));
```

Comment:
Just to confirm my understanding, this will create a concept set for each code (so we have a collection of concept sets each containing one code).

Reply:
Yeah, that's the intent. Is there another way to refer to single codes in a cohort definition without having a separate concept set for each one?

Reply:
No that's the right way to do it (AFAIK)!